### ISSUE TITLE
Error when plotting output signal from s.integrate_in_range()

### ISSUE THREAD

**Issue by magnunor** [hyperspy/hyperspy#996]
In the current master, running:

``` Python
s = hs.signals.Signal(np.arange(1000).reshape(100,10))
s_i = s.integrate_in_range()
# Select some area
s_i.plot()
```

Gives the error:

``` Python
<ipython-input-21-d328d94f2a72> in <module>()
----> 1 s_i.plot()

/home/magnunor/Documents/hyperspy/development/hyperspy/signal.py in plot(self, navigator, axes_manager, **kwargs)
   3342 
   3343         self._plot.plot(**kwargs)
-> 3344         self.events.data_changed.connect(self.update_plot, [])
   3345         if self._plot.signal_plot:
   3346             self._plot.signal_plot.events.closed.connect(

/home/magnunor/Documents/hyperspy/development/hyperspy/events.py in connect(self, function, kwargs)
    307         if function in self.connected:
    308             raise ValueError("Function %s already connected to %s." %
--> 309                              (function, self))
    310         if kwargs == 'auto':
    311             spec = inspect.signature(function)

ValueError: Function <bound method Signal.update_plot of <Signal, title: , dimensions: (10|)>> already connected to <hyperspy.events.Event: Event that triggers when the data has changed: {<bound method Signal.update_plot of <Signal, title: , dimensions: (10|)>>}>
```

The plot still appears, but the Error message is not so nice.

**Comment by thomasaarholt** [comment]
I keep getting this error on Mac as well, very often when doing things with plots of convoluted models.

**Comment by ericpre** [comment]
Fixed in #2631.
